### ISSUE TITLE
Make progress start/end single character

### ISSUE THREAD

**Issue by tonsky** [tonsky/FiraCode#1394]
@tonsky - I have an idea: you could add a ligature for the new FiraCode progress bar chars themselves, for the case where a bar is 100% completed. Some progress bar libraries allow setting a "theme" for the bar, with characters for bar start, filled progress, empty progress, and bar end, but do not have a concept of two different characters for bar end depending on whether the progress has reached 100% or not.

This results in a progress bar that looks slightly incomplete:

![image](https://user-images.githubusercontent.com/439499/158846201-22bc2583-b52d-47a6-8231-4ae28d6b6e07.png)

A ligature could solve that by replacing the sequence `\uee04\uee02` (filled bar, empty end) with `\uee04\uee05` (filled bar, filled end).

Even without all the application-specific ligatures, I think this would be useful.

_Originally posted by @joelnordell in https://github.com/tonsky/FiraCode/issues/1183#issuecomment-1071042997_

**Comment by tonsky** [comment]
@joelnordell yeah, that was an oversight on my part. Start/end should be the same always, I think

**Comment by joelnordell** [comment]
Perhaps the right solution is to introduce two _new_ characters as "auto-adapting start/end" along with ligatures that make them appear empty or filled based on the immediately-adjacent bar character (before for end cap, after for start cap). By making them new characters, existing behavior can be maintained for the existing chars, for any use cases where someone wants to explicitly control whether the ends are filled or not.

e.g.
![image](https://user-images.githubusercontent.com/439499/159013292-8d976706-85e6-4fd4-b87b-0209e899cc4f.png)
